### ISSUE TITLE
The example JS code doesn't call the server at the first call

### ISSUE THREAD

**Issue by pmontrasio** [rails/actioncable#118]
I experimented with ActionCable by building over the example code in the README. My goal was authenticating the websocket with devise, which I did, but along the way I run into a problem. I eventually traced it back to this code from the README

    $(document).on 'page:change', ->
      App.appearance.appear()

```App.appearance.appear()``` is called and returns before the websocket is open and connected to the server. It will always fail to trigger the ```appear``` action on the server unless something gives to the websocket enough time to connect to the server. In my case that time was provided by the breakpoints I set to inspect the issue, so I had the odd behavior of something working in the debugger and not without it. The long story is in the README at https://github.com/pmontrasio/actioncable-devise

TL;DR: I modified the example by adding a queue to store the calls until the websocket is connected and performing them after the ```connected``` callback is called. I did it as a proof of concept here https://github.com/pmontrasio/actioncable-devise/blob/master/app/assets/javascripts/channels/subscriptions/appearance.coffee but I think that won't scale.

A better solution could be building that kind of queue and replay mechanism in the ActionCable js code (maybe in ```Cable.Subscriptions```.)

However, it's strange that nobody run into this problem before with the very example on the home page of the gem so maybe I'm doing something wrong.

**Comment by kayakyakr** [comment]
Just wanted to write that you're not crazy, this does happen. I solved it by attaching a promise to the subscription connected message. The issue is that you really have two different points of failure: you can potentially send a message in the millisecond before the cable is connected, but you can also send a message before your subscription is connected. 

There is already handling if you try to send a message to a closed connection (as of now, it simply returns false). Sending to a subscription too early, before it has connected, is probably the bigger issue here.

I like the idea of attaching a queue to the send but that does run into some complications with the connected message: it should be handled inside the subscription class, but we still need to allow people to add a connected event without breaking the internals. Means we have to notify, say, _connected instead and have that pass on the notification for connected.

I might take a pass at it later today after I've done some real work.

**Comment by javan** [comment]
Subscriptions now receive the `connected` callback when their subscription is active on the server (https://github.com/rails/actioncable/pull/98) so a better example would be adding the `page:change` handler after `connected`.

```coffee
App.appearance = App.cable.subscriptions.create "AppearanceChannel",
  connected: ->
    @installEventHandlers()
    @appear()

  appear: ->
    ...
    
  installEventHandlers: ->
    unless @installedEventHandlers
      @installedEventHandlers = true
      $(document).on "page:change", =>
        @appear()
```

I'll take a pass at freshening up the examples.
